### PR TITLE
Switch to go modules

### PR DESCRIPTION
**Pull request by thaJeztah**
Splitting into two commits for a clearer history;

1. add go.mod and vendor upstream pkg/errors v0.8.0

    This patch replaces the vendor.conf with a go.mod, and re-vendors the pkg/errors files from upstream. This commit is mostly to preserve history of the local changes that were made to the pkg/errors package (which appear to have been made for debugging purposes). Setting the minimum go version in go.mod to v1.7 for this commit, which matches the lowest version that's testet against in CI.

2. go.mod: update pkg/errors v0.9.1 and remove local vendor

    raising the minimum go version in go.mod 1.12, to fetch the dependency module. Updated travis to test against 1.12 (minimum version) and 1.15 (current stable).

I'll also open a follow-up to remove the dependency on pkg/errors, however that requires Go 1.13 or up, so perhaps you'd want to keep that separate (and tag a separate version for that change)

### CODE REVIEW

**Comment by thaJeztah** [comment]
@cyphar @kolyshkin PTAL

(follow-up is in https://github.com/cyphar/filepath-securejoin/pull/4)

**Comment by kolyshkin** [comment]
So, why go 1.12 and not, say, 1.14 (the oldest currently supported release), or 1.13 (the most fresh unsupported release)?

**Comment by thaJeztah** [comment]
> So, why go 1.12 and not, say, 1.14 

Trying to keep the go.mod to reflect the actual minimum required version of Go (although I don't think go modules so far _enforce_ the minimum version specified)

File: vendor/github.com/pkg/errors/errors.go [hunk 1]
```diff
@@ -1,310 +0,0 @@
...
-	return value
-}
-
-// Debug sets whether or not debugging is enabled. If enabled, then formatting
-// an error wrapped by pkg/error with "%v" will return a full stack trace. This
-// allows you to dynamically decide whether to output stack traces (without
-// having to use "%+v" indiscriminately).
-func Debug(value bool) {
-	debugEnabled.Set(value)
-}
```
Comment (cyphar) [inline vendor/github.com/pkg/errors/errors.go:138]: Not sure if you noticed this, but this is actually a patched version of `pkg/error`. Though we don't *really* need this anymore, so no worries.

Comment (thaJeztah) [reply vendor/github.com/pkg/errors/errors.go:138]: Yes, I noticed; I did the first commit separately for that;

> This commit is mostly to preserve history of the local changes that were made to the pkg/errors package (which appear to have been made for debugging purposes). (...)

Comment (cyphar) [reply vendor/github.com/pkg/errors/errors.go:138]: Sorry, my fault for not reading :bow:.

Comment (thaJeztah) [reply vendor/github.com/pkg/errors/errors.go:138]: No worries! It was a good catch 👍

**Comment by cyphar** [comment]
Closing in favour of #7.

**Pull request closed without merging**